### PR TITLE
[SPARK-31763][PYSPARK] Add `inputFiles` method in PySpark DataFrame Class.

### PR DESCRIPTION
**Pull request by iRakson**
### What changes were proposed in this pull request?
Adds `inputFiles()` method to PySpark `DataFrame`. Using this, PySpark users can list all files constituting a `DataFrame`.

**Before changes:**

```
>>> spark.read.load("examples/src/main/resources/people.json", format="json").inputFiles()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/***/***/spark/python/pyspark/sql/dataframe.py", line 1388, in __getattr__
    "'%s' object has no attribute '%s'" % (self.__class__.__name__, name))
AttributeError: 'DataFrame' object has no attribute 'inputFiles'
```

**After changes:**

```
>>> spark.read.load("examples/src/main/resources/people.json", format="json").inputFiles()
[u'file:///***/***/spark/examples/src/main/resources/people.json']
```

### Why are the changes needed?
This method is already supported for spark with scala and java.

### Does this PR introduce _any_ user-facing change?
Yes, Now users can list all files of a DataFrame using `inputFiles()`

### How was this patch tested?
UT added.

### CODE REVIEW

**Comment by iRakson** [comment]
cc @HyukjinKwon

File: python/pyspark/sql/tests/test_dataframe.py [hunk 1]
```diff
@@ -820,6 +822,21 @@ def test_same_semantics_error(self):
             with self.assertRaisesRegexp(ValueError, "should be of DataFrame.*int"):
                 self.spark.range(10).sameSemantics(1)
 
+    def test_input_files(self):
+        tmpPath = tempfile.mkdtemp()
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_dataframe.py:826]: Let's to a try-finally, and use `this_naming_rule` per PEP8.

```python
tpath = tempfile.mkdtemp()
shutil.rmtree(tpath)
try:
    ...
finally:
    shutil.rmtree(tpath)
```

File: python/pyspark/sql/dataframe.py [hunk 2]
```diff
@@ -2219,6 +2219,20 @@ def semanticHash(self):
...
+        Returns a best-effort snapshot of the files that compose this :class:`DataFrame`.
+        This method simply asks each constituent BaseRelation for its respective files and
+        takes the union of all results. Depending on the source relations, this may not find
+        all input files. Duplicates are removed.
+
+        >>> df = spark.read.load("examples/src/main/resources/people.json", format="json")
+        >>> len(df.inputFiles())
+        1
+        """
+        return [f for f in self._jdf.inputFiles()]
```
Comment (HyukjinKwon) [inline python/pyspark/sql/dataframe.py:2234]: You can just `return list(self._jdf.inputFiles())`

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by iRakson** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by iRakson** [comment]
Thank You. @HyukjinKwon 
Is it a good idea to add this method in 3.0 as well ?

**Comment by HyukjinKwon** [comment]
Technically this is a new API and we shouldn't backport it.

**Pull request closed without merging**